### DISCUSSION TITLE
Welcome to php-examples-for-aws-lambda Discussions!

### DISCUSSION THREAD

**Discussion by benjasl-stripe** [aws-samples/php-examples-for-aws-lambda#18] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by austingrove** [comment]
Hello,
Thanks for making this tutorial. 
I have zipped the php and bootstrap file in a cloud9 environment. I am on "Creating Dependencies".

1. The text says "In an Amazon Linux environment compatible with the Lambda execution environment", can you clarify on this?
2. How will I be sending the package out of cloud9 and make useful for lambda? Is that what the "Publish to Lambda layers" commands do? Do I need to edit the commands at all?

**Reply by benjasl-stripe** [reply]
1. Essentially it means launching and compiling PHP in an EC2 instance that uses the same Amazon Machine Image (AMI) from the underlying Lambda Execution Environment. If you are using a cloud 9 environment then that should be sufficient, this blog post has more info on the particulars : https://aws.amazon.com/blogs/apn/aws-lambda-custom-runtime-for-php-a-practical-example/
   2.  Once you have zipped the runtime up (in the directory structure shown) you will use the AWS CLI to publish this ZIP file to a Lambda Layer. you will then configure your PHP Lambda function to use this Layer. You shouldn't need to change the commands.

**Reply by austingrove** [reply]
This helps immensely. Thank you so much
